src/garage/torch/embeddings/recurrent_encoder.py

```diff
@@ -0,0 +1,110 @@
+# pylint: disable=attribute-defined-outside-init
+"""A recurrent network with LSTM for encoding context of RL tasks."""
+
+import torch
+from torch import nn
+
+from garage.torch.modules import MLPModule
+
+
+class RecurrentEncoder(MLPModule):
+    """This recurrent network encodes context of RL tasks.
+
+    Context is stored in the terms of observation, action, and reward, and this
+    network uses an MLP module followed by an LSTM model for encoding it.
+
+    Args:
+        *args: MLPModule arguments.
+        **kwargs: MLPModule arguments including:
+            input_dim (int) : Dimension of the network input.
+            output_dim (int): Dimension of the network output.
+            hidden_sizes (list[int]): Output dimension of dense layer(s).
+                For example, (32, 32) means this MLP consists of two
+                hidden layers, each with 32 hidden units.
+            hidden_nonlinearity (callable or torch.nn.Module): Activation
+                function for intermediate dense layer(s). It should return a
+                torch.Tensor.Set it to None to maintain a linear activation.
+            hidden_w_init (callable): Initializer function for the weight
+                of intermediate dense layer(s). The function should return a
+                torch.Tensor.
+            hidden_b_init (callable): Initializer function for the bias
+                of intermediate dense layer(s). The function should return a
+                torch.Tensor.
+            output_nonlinearity (callable or torch.nn.Module): Activation
+                function for output dense layer. It should return a
+                torch.Tensor. Set it to None to maintain a linear activation.
+            output_w_init (callable): Initializer function for the weight
+                of output dense layer(s). The function should return a
+                torch.Tensor.
+            output_b_init (callable): Initializer function for the bias
+                of output dense layer(s). The function should return a
+                torch.Tensor.
+            layer_normalization (bool): Bool for using layer normalization or
+                not.
+
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._hidden_dim = self._hidden_sizes[-1]
+        # hidden dimension should be (task, 1, feat)
```

Comment:
hmm if this is essential, considering checking and raising an exception or asserting.

use an assert if the rest of your code should have already assured this

use an exception if incorrect user input could cause this to happen.